plugins/modules/zos_blockinfile.py

```diff
@@ -569,26 +610,35 @@ def main():
     # state=present, insert/replace a block with matching regex pattern
     # state=absent, delete blocks with matching regex pattern
     if parsed_args.get('state') == 'present':
-        return_content = present(src, block, marker, ins_aft, ins_bef, encoding, force)
+        if '"' in block:
+            rc, cmd = execute_dmod(src, block, quotedString(marker), force, encoding, module=module, ins_bef=quotedString(ins_bef),
```

Comment:
I think this would yield an inconsistent result when having these two inputs : 
```
block: 'This is a double quote " '
ins_bef: ' Insert before this " double quote " ' 
```
Than this where I removed the double quotation mark from the block option:

```
block: 'This is a double quote'
ins_bef: ' Insert before this " double quote " ' 
```
Assuming the text in the data set is: 

```
 Insert before this " double quote " 
```
That is because when block contains a `"` symbol it will remove any `"` from `ins_bef`, `ins_aft` and `marker`, but it won't do the same if block does not contain `"`. Could you test this scenario?

Reply:
resolved